src/Appwrite/Platform/Modules/Functions/Workers/Builds.php

```diff
@@ -473,16 +478,37 @@ protected function buildDeployment(Device $deviceForFunctions, Func $queueForFun
                 $vars[$var->getAttribute('key')] = $var->getAttribute('value', '');
             }
 
-            $cpus = $spec['cpus'] ?? APP_FUNCTION_CPUS_DEFAULT;
-            $memory = max($spec['memory'] ?? APP_FUNCTION_MEMORY_DEFAULT, 1024); // We have a minimum of 1024MB here because some runtimes can't compile with less memory than this.
+            $collection = $resource->getCollection();
 
-            $jwtExpiry = (int)System::getEnv('_APP_FUNCTIONS_BUILD_TIMEOUT', 900);
-            $jwtObj = new JWT(System::getEnv('_APP_OPENSSL_KEY_V1'), 'HS256', $jwtExpiry, 0);
-            //todo: not needed for sites yet, might be useful as a build variable but too shortlived
-            $apiKey = $jwtObj->encode([
-                'projectId' => $project->getId(),
-                'scopes' => $resource->getAttribute('scopes', [])
-            ]);
+            $cpus = match ($collection) {
+                'functions' => $spec['cpus'] ?? APP_FUNCTION_CPUS_DEFAULT,
+                'sites' => $siteVars['cpus'] ?? APP_SITE_CPUS_DEFAULT,
+                default => APP_FUNCTION_CPUS_DEFAULT,
```

Comment:
i dont think we need a default 👍🏻 it will always be one of those, otherwise we throw at the beginning